File: ios/Parsers/GeolocationParser.swift
import Foundation
import Crisp

@objc
public class GeolocationParser: NSObject {

    @objc
    public static func fromDictionary(_ dict: NSDictionary) -> Geolocation {
        let country = dict.getStringOrNil(forKey: "country")
        let city = dict.getStringOrNil(forKey: "city")
        return Geolocation(city: city, country: country)
    }
}

File: android/src/main/java/com/reactnativecrispchatsdk/parsers/GeolocationParser.kt
package com.reactnativecrispchatsdk.parsers

import com.facebook.react.bridge.ReadableMap
import com.reactnativecrispchatsdk.getStringOrNull
import im.crisp.client.external.data.Geolocation

/**
 * Parser for Geolocation objects from JavaScript
 * Converts ReadableMap to Crisp SDK Geolocation class
 *
 * TypeScript interface:
 * interface Geolocation {
 *   country?: string;
 *   city?: string;
 * }
 */
object GeolocationParser {
    fun fromReadableMap(map: ReadableMap): Geolocation {
        val country = map.getStringOrNull("country")
        val city = map.getStringOrNull("city")
        return Geolocation(country, city)
    }
}
